app/controllers/alchemy/admin/contents_controller.rb

```diff
@@ -32,9 +32,10 @@ def update
       end
 
       def order
-        params[:content_ids].each do |id|
-          content = Content.find(id)
-          content.move_to_bottom
+        Content.transaction do
+          params[:content_ids].each_with_index do |id, idx|
+            Content.where(id: id).update_all(position: idx + 1)
```

Comment:
I can't test right now, as I'm on a bus. I am, however, fine with the
current code if it produces less queries. :beer:
On Jan 23, 2016 12:20 PM, "Thomas von Deyen" <notifications@github.com>
wrote:

> In app/controllers/alchemy/admin/contents_controller.rb
> <https://github.com/AlchemyCMS/alchemy_cms/pull/918#discussion_r50623483>:
>
> > @@ -32,9 +32,10 @@ def update
> >        end
> >
> >        def order
> > -        params[:content_ids].each do |id|
> > -          content = Content.find(id)
> > -          content.move_to_bottom
> > +        Content.transaction do
> > +          params[:content_ids].each_with_index do |id, idx|
> > +            Content.where(id: id).update_all(position: idx + 1)
>
> If this only produces one SQL query, then I'm fine with it. Otherwise we
> need to live with that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AlchemyCMS/alchemy_cms/pull/918/files#r50623483>.
>